### ISSUE TITLE
Share Plugin raising err when trying to export from Godot 4.2.1

### ISSUE THREAD

**Issue by x-mugen-x** [cengiz-pz/godot-ios-share-plugin#3]
Hello,
After installing your plugin through Asset Lib,
I enable it in Project settings and in the Export template for iOS.
Without even adding the Share node, when I try to export my project I get these errors (only when the share plugin is enabled).
```
platform/ios/export/export_plugin.cpp:1225 - Condition "err" is true. Returning: err
platform/ios/export/export_plugin.cpp:1241 - Condition "err" is true. Returning: err
editor/export/editor_export_platform.h:182 - Xcode Build: Xcode project build failed, see editor log for details.
```
I've used your Android share plugin but never had this issue (great plugins by the way).
Any help would be appreciated,
Thank you

**Comment by cengiz-pz** [comment]
Hi @x-mugen-x, in the plugin's `.gdip` file, could you replace the following line

    binary="SharePlugin.release.a"

with

    binary="SharePlugin.a"

& let me know if that solves your issue?

**Comment by x-mugen-x** [comment]
> Hi @x-mugen-x, in the plugin's `.gdip` file, could you replace the following line
> 
> ```
> binary="SharePlugin.release.a"
> ```
> 
> with
> 
> ```
> binary="SharePlugin.a"
> ```
> 
> & let me know if that solves your issue?

Thanks for the swift reply.
I did as you suggested, I still get the first 2 err lines during compilation, but the project compiles now, I don't get the final error line.
Will this work though? There is no `SharePlugin.a` in the plugin folder only `SharePlugin.debug.a` and `SharePlugin.release.a`

**Comment by x-mugen-x** [comment]
I missed a click, but yes please let me know regarding the remaining `err` lines.
Thank you

**Comment by cengiz-pz** [comment]
> Will this work though? There is no SharePlugin.a in the plugin folder only SharePlugin.debug.a and SharePlugin.release.a

Yes, apparently, the engine handles it.  (_The engine points to the debug library for debug builds and release lib for release builds._)

> I missed a click, but yes please let me know regarding the remaining err lines.

I would need more information from your side.  Could you attach a minimal project that produces the error messages?

**Comment by x-mugen-x** [comment]
> > Will this work though? There is no SharePlugin.a in the plugin folder only SharePlugin.debug.a and SharePlugin.release.a
> 
> Yes, apparently, the engine handles it. (_The engine points to the debug library for debug builds and release lib for release builds._)
> 
> > I missed a click, but yes please let me know regarding the remaining err lines.
> 
> I would need more information from your side. Could you attach a minimal project that produces the error messages?

Hmmm... have you tested it in an 4.2.1 iOS project? I just tried adding the Share node to my scene and am getting the `SharePlugin singleton not found` error. I have it enabled in project settings and the export template though.
I can try reinstalling it from the release archive instead of AssetLib if you think that might help.
Thanks

**Comment by cengiz-pz** [comment]
> Hmmm... have you tested it in an 4.2.1 iOS project?
Yes

> I just tried adding the Share node to my scene and am getting the SharePlugin singleton not found error. I have it enabled in project settings and the export template though.

What device are you running your app on?  Are there and errors or warnings in the logs?

> I can try reinstalling it from the release archive instead of AssetLib if you think that might help.

If the Share checkbox is visible in the `Plugins` section of `Project Settings` and in the `iOS Export Settings`, then it should be installed properly.

**Comment by cengiz-pz** [comment]
> Hmmm... have you tested it in an 4.2.1 iOS project?

Yes

> I just tried adding the Share node to my scene and am getting the SharePlugin singleton not found error. I have it enabled in project settings and the export template though.

What device are you running your app on?  Are there and errors or warnings in the logs?

> I can try reinstalling it from the release archive instead of AssetLib if you think that might help.

If the Share checkbox is visible in the `Plugins` section of `Project Settings` and in the `iOS Export Settings`, then it should be installed properly.

**Comment by cengiz-pz** [comment]
> Hmmm... have you tested it in an 4.2.1 iOS project?

Yes

> I just tried adding the Share node to my scene and am getting the SharePlugin singleton not found error. I have it enabled in project settings and the export template though.

What device are you running your app on?  Are there any errors or warnings in the logs?

> I can try reinstalling it from the release archive instead of AssetLib if you think that might help.

If the Share checkbox is visible in the `Plugins` section of `Project Settings` and in the `iOS Export Settings`, then it should be installed properly.

**Comment by x-mugen-x** [comment]
I tried completely removing all traces of the plugin and installing the release version as well as setting the gdip to the debug version explicitely and a few other things inlcuding autoload but kept getting the singleton not found error. The only thing I can think of is that my project was branched from an Android project that was using your Android share plugin and although that was completely removed, perhaps there was a conflicting temp file somewhere.
In the end I incorportated your share function directly without using the plugin install and it works as expected.
I'll close this issue as I'm not sure where else to troubleshoot this one.
Thanks all the same.
